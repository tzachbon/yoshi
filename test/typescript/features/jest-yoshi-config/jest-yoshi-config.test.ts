import Scripts from '../../../scripts';

const scripts = Scripts.setupProjectFromTemplate({
  templateDir: __dirname,
  projectType: 'typescript',
});

jest.setTimeout(35000);

describe.each(['prod', 'dev'] as const)('e2e [%s]', (mode) => {
  it('run tests', async () => {
    await scripts[mode](async () => {
      try {
        await scripts.test(mode);
        throw new Error('Test should have failed but passed');
      } catch (e) {
        expect(e.message).toMatchSnapshot();
      }
    });
  });
});
