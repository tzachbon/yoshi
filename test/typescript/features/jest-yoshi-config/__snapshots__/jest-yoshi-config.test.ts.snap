// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e [dev] run tests 1`] = `
"[13:09:32] Starting 'cdn'...
	cdn is already running on 3200, skipping...
[13:09:34] Finished 'cdn' after 1953 ms

        When using server.command in jest config you must specify your port.

        For example:
        
        module.exports = {
          server: {
            command: 'node dist/server.js',
            port: 3100,
          },
        }
jest failed with status code \\"1\\""
`;

exports[`e2e [prod] run tests 1`] = `
"[13:09:23] Starting 'cdn'...
	cdn is already running on 3200, skipping...
[13:09:24] Finished 'cdn' after 651 ms

        When using server.command in jest config you must specify your port.

        For example:
        
        module.exports = {
          server: {
            command: 'node dist/server.js',
            port: 3100,
          },
        }
jest failed with status code \\"1\\""
`;
